### PR TITLE
chore(playwrighttesting): removed fallback authentication logic

### PR DESCRIPTION
**Pull request by kashish2508**
### Packages impacted by this PR

@azure/microsoft-playwright-testing

### Issues associated with this PR

Remove fallback logic in authentication
Improved validation of Pat token.

### Describe the problem that is addressed by this PR

Earlier, we used to have fallback logic in authentication. Entra -> PAT
or PAT -> Entra. This PR removes the fallback logic.

### CODE REVIEW

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/PlaywrightService.cs [hunk 1]
```diff
@@ -107,9 +108,9 @@ internal PlaywrightService(OSPlatform? os = null, string? runId = null, string?
         // fetch Entra id access token if required
         // 1. Entra id access token has been fetched once via global functions
         // 2. Not close to expiry
-        if (!string.IsNullOrEmpty(_entraLifecycle!._entraIdAccessToken) && _entraLifecycle!.DoesEntraIdAccessTokenRequireRotation())
+        if (!string.IsNullOrEmpty(_entraLifecycle!._entraIdAccessToken) && _entraLifecycle!.DoesEntraIdAccessTokenRequireRotation() && ServiceAuth == ServiceAuthType.EntraId)
```
Comment (Sid200026) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/PlaywrightService.cs:111]: _entraLifecycle!._entraIdAccessToken is always entra token. So the additional condition of checking auth type is not required.

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/PlaywrightService.cs [hunk 2]
```diff
@@ -254,14 +246,23 @@ private void ValidateMptPAT()
         try
         {
             string authToken = GetAuthToken()!;
+            if (string.IsNullOrEmpty(authToken))
+                throw new Exception(Constants.s_no_auth_error);
             JsonWebToken jsonWebToken = _jsonWebTokenHandler!.ReadJsonWebToken(authToken) ?? throw new Exception(Constants.s_invalid_mpt_pat_error);
+            var tokenaWorkspaceId = jsonWebToken.Claims.FirstOrDefault(c => c.Type == "aid")?.Value;
```
Comment (Sid200026) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/PlaywrightService.cs:252]: NIT: tokenWorkspaceId

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/PlaywrightService.cs [hunk 3]
```diff
@@ -254,14 +246,23 @@ private void ValidateMptPAT()
...
+            Match match = Regex.Match(ServiceEndpoint, @"wss://(?<region>[\w-]+)\.api\.(?<domain>playwright(?:-test|-int)?\.io|playwright\.microsoft\.com)/accounts/(?<workspaceId>[\w-]+)/");
+            if (!match.Success)
+                throw new Exception(Constants.s_invalid_service_endpoint_error_message);
+            var serviceEndpointworkspaceId = match.Groups["workspaceId"].Value;
```
Comment (Sid200026) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/PlaywrightService.cs:256]: NIT: serviceEndpointWorkspaceId

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/PlaywrightService.cs [hunk 4]
```diff
@@ -254,14 +246,23 @@ private void ValidateMptPAT()
...
+            if (tokenaWorkspaceId != serviceEndpointworkspaceId)
+                throw new Exception(Constants.s_workspace_mismatch_error);
             var expiry = (long)(jsonWebToken.ValidTo - new DateTime(1970, 1, 1)).TotalSeconds;
             if (expiry <= DateTimeOffset.UtcNow.ToUnixTimeSeconds())
                 throw new Exception(Constants.s_expired_mpt_pat_error);
         }
         catch (Exception)
```
Comment (Sid200026) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/PlaywrightService.cs:263]: No need to catch, if we are simply throwing the Exception again.

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/tests/EntraLifecycleTests.cs [hunk 5]
```diff
@@ -132,7 +132,7 @@ public async Task FetchEntraIdAccessTokenAsync_WhenTokenIsFetched_SetsTokenAndEx
     }
```
Comment (Sid200026) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/tests/EntraLifecycleTests.cs:132]: There will be some tests that checks the fallback logic. Modify those tests to ensure fallback is no longer happening.

**Review by Sid200026 — APPROVED** [review]
LGTM, once the broken tests are fixed, feel free to merge this PR.

**Review by puagarwa — APPROVED** [review]

**Pull request merged**